common/oatbox/filesystem/FileSystemService.php

```diff
@@ -69,14 +95,16 @@ public function hasDirectory($id)
      * Retrieve an existing FileSystem by ID.
      *
      * @param string $id
-     * @return FileSystem
+     * @return FilesystemInterface
      * @throws \common_exception_Error
      * @throws \common_exception_NotFound
      */
     public function getFileSystem($id)
     {
         if (!isset($this->filesystems[$id])) {
-            $this->filesystems[$id] = new FileSystem($id, $this->getFlysystemAdapter($id));
+            $config = $this->getAdapterConfig($id);
```

Comment:
Same as above, this is a (slightly functionality extended) repository. Can we make it a separate object used by the service?